javatools/__init__.py

```diff
@@ -2093,10 +2098,7 @@ def _pretty_const_type_val(typecode, val):
 
     if typecode == CONST_Utf8:
         typestr = "Utf8"  # formerly Asciz, which was considered Java bug
-        if isinstance(val, unicode):
-            val = repr(val)[2:-1]  # trim off the surrounding u"" (HACK)
-        else:
-            val = repr(val)[1:-1]  # trim off the surrounding "" (HACK)
+        val = val.encode('utf-8')
```

Comment:
the point of spitting out a repr was to escape certain characters, not just to encode.

Reply:
* Dependency on `future` and `past`:

In fact `past` can be avoided. (By replacing all `xrange` with `range`; this does not really have an impact on Py2, as our input data is small - definitely not at size where difference between generator and list could be noticed. On Py3, `range` is the only one left (I like [this writing](http://treyhunner.com/2018/02/python-3-s-range-better-than-python-2-s-xrange/) on the subject)).

I have amended the commit where I had imported `past`, rebased and force-pushed the branch. 
(BTW, **every** commit of this rather large push request passes **all** tests in Py2.)

Reply:
_> the point of spitting out a repr was to escape certain characters, not just to encode._

Reverted back to what you had, but in a form which compiles under Py3. Commit amended, branch force-pushed. (Please tell if you find force-pushing of PRs' branches inconvenient.)

[2.7 and 3.6 tests pass](https://travis-ci.org/KonstantinShemyak/python-javatools/builds/436706619), although these lines were obviously not covered.

In fact our Unicode support is not tested. I had earlier https://github.com/KonstantinShemyak/python-javatools/issues/34 and just created https://github.com/KonstantinShemyak/python-javatools/issues/42 and https://github.com/KonstantinShemyak/python-javatools/issues/43. But I want to get Py3 otherwise supported before checking these issues, so that they would be handled in both language versions.

Reply:
For cleaner history, I have squashed the latter change into the commit with message "Python3-compatible syntax", so there is no incorrect code in any commit in the chain.